File: algorithms/1704.determine-if-string-halves-are-alike/solution.rs
impl Solution {
    pub fn halves_are_alike(s: String) -> bool {
        let vowels: Vec<char> = vec!['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];

        let mid = s.len() / 2;
        let mut a_vowels = 0;
        let mut b_vowels = 0;

        for (i, c) in s.chars().enumerate() {
            if vowels.contains(&c) {
                if i < mid {
                    a_vowels += 1
                } else {
                    b_vowels += 1
                }
            }
        }

        a_vowels == b_vowels
    }
}

File: algorithms/1704.determine-if-string-halves-are-alike/solution.ts
function halvesAreAlike(s: string): boolean {
  const vowels = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"];
  const mid = s.length / 2;

  let aVowels = 0;
  let bVowels = 0;
  for (let i = 0; i < s.length; i++) {
    if (vowels.includes(s[i])) {
      if (i < mid) {
        aVowels++;
      } else {
        bVowels++;
      }
    }
  }

  return aVowels == bVowels;
}
